### PR TITLE
Correctly handle files with duplicate picture ids. XLPictures refactoring

### PR DESCRIPTION
**Pull request by Pankraty**
This PR partially addresses #808 

1. 2 files are invalid (`formats.xlsx` and `tdf100709.xlsx`). ClosedXML is not designed to manage corrupted files so I suggest removing them.
2. `xf_default_values.xlsx` has pictures with the duplicate ids. Apparently, duplicate IDs are allowed in XLSX so I removed the validation that threw exceptions on attempts to assign the existing ID to the picture.

### CODE REVIEW

**Comment by kenleese** [comment]
@Pankraty Should ClosedXml have a deterministic behavior for invalid / corrupted files? If so, those files could be the basis of a useful test case.

File: ClosedXML/Excel/Drawings/XLPicture.cs [hunk 1]
```diff
@@ -140,17 +139,7 @@ public Int32 Height
...
-            get { return id; }
-            internal set
-            {
-                if ((_worksheet.Pictures.FirstOrDefault(p => p.Id.Equals(value)) ?? this) != this)
-                    throw new ArgumentException($"The picture ID '{value}' already exists.");
-
-                id = value;
-            }
-        }
+        public Int32 Id { get; internal set; }
```
Comment (igitur) [inline ClosedXML/Excel/Drawings/XLPicture.cs]: Private set?

**Comment by igitur** [comment]
I remember how surprised I was to find out picture names were not unique. Now I'm even more surprised that picture IDs are not unique. What's the point of having an ID if it's not unique.

Anyway, compare this to #819, which is about a fix for having unique IDs for the non visual drawing properties. I

**Comment by igitur** [comment]
For the corrupt files, does OpenXML throw an exception? If so we could handle it. But otherwise it's difficult to tell whether a loading problem will be due to a corrupt file or a ClosedXML bug.

**Comment by vbjay** [comment]
I'd just st as well fix it in load. When a duplicate I'd is found, replace
it with a valid one.

File: ClosedXML/Excel/Drawings/XLPicture.cs (hunk 1 above)
Comment (Pankraty) [reply ClosedXML/Excel/Drawings/XLPicture.cs]: With `private set`, we won't be able to assign the value read from a file.

Comment (igitur) [reply ClosedXML/Excel/Drawings/XLPicture.cs]: Sorry, stupid me.

**Comment by igitur** [comment]
Please run those files through the validator first.

**Comment by Pankraty** [comment]
The validator says "Attribute 'id' should have unique value. Its current value '1' duplicates with others".

**Comment by vbjay** [comment]
Question. Does there need to be any relationship fixing?

**Comment by igitur** [comment]
This changes the names of pictures if there are duplicates. This could possibly break embedded VBA code, e.g. `ActiveSheet.Shapes("Picture 2").IncrementLeft 30`. If it's a choice between supporting files that LibreOffice can load/save and not breaking Excel VBA code, I think I'm leaning towards keeping the VBA code safe.

**Comment by Pankraty** [comment]
> This could possibly break embedded VBA code,

The thing is, when such file is opened in Excel, the pictures with duplicate names receive new ones, so we won't make it worse.

**Comment by igitur** [comment]
@Pankraty Are you sure? In #444 there is an attached file with images with duplicate names `Picture 2`. If I open this in Excel and resave it, the names remain `Picture 2`.

**Comment by Pankraty** [comment]
You are right. For `xf_default_values.xlsx` it behaves differently but probably that's because in this file pictures have the same IDs as well.

Shall we park it for now?

**Comment by igitur** [comment]
Or maybe because in #444 's file the pictures are inside groups. But even the groups' names are duplicates. Yes, let's park.

**Pull request closed without merging**